react-components/src/components/NodeCacheProvider/Image360AnnotationCache.ts

```diff
@@ -60,25 +66,59 @@ export class Image360AnnotationCache {
 
     const assetsArray = await fetchAssetForAssetIds(Array.from(filteredAssetIds), this._sdk);
     const assets = new Map(assetsArray.map((asset) => [asset.id, asset]));
+    const assetsWithAnnotations = await this.getAssetWithAnnotationsMapped(annotationsInfo, assets);
 
-    const assetsWithAnnotations: Image360AnnotationAssetInfo[] = annotationsInfo.reduce(
-      (acc: Image360AnnotationAssetInfo[], annotationInfo) => {
+    return assetsWithAnnotations;
+  }
+
+  private async getAssetWithAnnotationsMapped(
+    annotationsInfo: AssetAnnotationImage360Info[],
+    assets: Map<number, Asset>
+  ): Promise<Image360AnnotationAssetInfo[]> {
+    const assetsWithAnnotationsPromises = annotationsInfo
+      .filter((annotationInfo) => {
+        const assetId = annotationInfo?.annotationInfo?.data?.assetRef?.id;
+        return assetId !== undefined && assets.has(assetId);
+      })
+      .map(async (annotationInfo) => {
         const assetId = annotationInfo.annotationInfo.data.assetRef.id;
-        if (assetId !== undefined && assets.has(assetId)) {
-          const asset = assets.get(assetId);
-          if (asset === undefined) {
-            return acc;
+        if (assetId === undefined) {
+          return undefined;
+        }
+        const asset = assets.get(assetId);
+        if (asset === undefined) {
+          return undefined;
+        }
+        const transform = annotationInfo.imageEntity.transform;
+        const revisionAnnotations = await annotationInfo.imageRevision.getAnnotations();
+
+        const centerPosition = new Vector3();
+
+        for (const revisionAnnotation of revisionAnnotations) {
+          if (
+            assetId !== getAssetIdOrExternalIdFromImage360Annotation(revisionAnnotation.annotation)
+          ) {
+            continue;
+          }
+          const center = revisionAnnotation.getCenter().applyMatrix4(transform);
+          console.log('assetId, center', assetId, center);
+          if (center !== undefined) {
+            centerPosition.add(center);
```

Comment:
Do you mean to use `add` here? Don't we only want to return the `centerPosition` of the `revisionAnnotation` that has the same ID as `annotationInfo.annotation`?